airflow/api_fastapi/execution_api/routes/task_instances.py

```diff
@@ -151,14 +151,16 @@ def ti_run(
                 "previous_state": previous_state,
             },
         )
-    log.info("Task with %s state started on %s ", previous_state, ti_run_payload.hostname)
+    else:
+        log.info("Task with %s state started on %s ", previous_state, ti_run_payload.hostname)
     # Ensure there is no end date set.
     query = query.values(
         end_date=None,
         hostname=ti_run_payload.hostname,
         unixname=ti_run_payload.unixname,
         pid=ti_run_payload.pid,
-        state=State.RUNNING,
+        state=TaskInstanceState.RUNNING,
+        last_heartbeat_at=timezone.utcnow(),
```

Comment:
so we're setting last_heartbeat_at now on the transition to run? this should fix the issue that https://github.com/apache/airflow/pull/46777 is attempting to fix too

Reply:
Yeah, I almost think it's reasonable to treat any request to the Exec API form a client as a heartbeat -- so requesting XCom, a connection etc updates the heartbeat, since we know it was alive.

We don't do it yet on those, but having said that out load (so to speak) I think that makes sense.

Reply:
yeah reasonable. i'll close the other PR in favor of this then